_posts/2020-09-08-kafka-rest-client.adoc

```diff
@@ -0,0 +1,203 @@
+---
+layout: post
+title: 'Combining Apache Kafka and the Rest client'
+date: 2020-09-08
+tags: reactive kafka rest mutiny http
+synopsis: How to combine reactive messaging, Kafka and the Rest client
+author: cescoffier
+---
+
+Another week, another interesting question. 
+This week, someone asks me about combining Kafka and the Rest Client. 
+That is a recurring subject, and most of the time, the goal is to achieve the following process:
+
+image:/assets/images/posts/kafka-rest-client/kafka-rest-architecture.png[]
+
+In other words, we want to call a remote service for each Kafka message we receive. 
+Therefore, we have a first topic ("in") containing the data we are consuming, for instance, "transactions". 
+Then, we have the central piece of the architecture: the processing component. 
+It consumes the incoming transactions, and for each of them, calls a remote service. 
+It also writes the response (produced by the remote service) to another Kafka topic "out".
+
+Implementing this with Quarkus is straightforward, and that's what we will cover in this post.
+Thanks to Reactive Messaging and the Rest Client, this should not take more than 20 lines of code!
+
+== The Remote Service
+
+Let's start with the remote service. 
+Quarkus offers multiple ways to invoke a remote HTTP service, but let's use the Rest Client as it provides an excellent way to interact with HTTP services without having to handle the low-level details of HTTP.
+
+You can use any HTTP API, but to simplify, let's consider a straightforward remote service, something like:
+
+[source, java]
+----
+@RegisterRestClient(configKey = "transaction-service")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
```

Comment:
So your using the full URL in config rather than adding a `@Path(/transactions)` here?

That's ok, just wanted to clarify

Reply:
Might be better, let me change

Reply:
Fixed

Reply:
I was thinking about adding it to the class, but having it on each method does make it clear what's happening